### ISSUE TITLE
App crashes after ARSession.Reset (only iOS)

### ISSUE THREAD

**Issue by EVOsparkLP** [Azure/azure-spatial-anchors-samples#357]
Hi,

my app crashes when ARSession.Reset() is called and a CloudManager.Session was running before. If the reset is called before the CloudManager.Session was running everything is fine.
Destroying or Resetting the CloudManager.Session before the ARSession.Reset() does not help. This issue was introduced with ASA 2.11.0. Until 2.10.2 the reset worked fine. 
The issue occurs only on iOS devices - no problems on Android.

**Steps to reproduce the issue**

1. Get the Azure sample scene "AzureSpatialAnchorsBasicDemo"
2. Modify the state "DemoStepStopSession" like this (based on suggestions from #284)
```case AppState.DemoStepStopSession:
    currentAppState = AppState.DemoStepBusy;
    CloudManager.StopSession();
    CleanupSpawnedObjects();

    // orig:
    // await CloudManager.ResetSessionAsync();

    // new:
    CloudManager.DestroySession();
    ARSession arSession = FindObjectOfType<ARSession>();
    arSession.Reset();
    //

    currentAppState = AppState.DemoStepCreateSessionForQuery;
    break;
```

3. Run the app on an iOS device
4. App crashes during the StopSession step

Here's my configuration:
Unity 2020.3.34
ARFoundation 4.1.10 & 4.2.3
ASA 2.11.0 & 2.12.0
The issue does NOT occur with ASA 2.9.0 and 2.10.2.
iPad 11 Pro 2020

**Comment by Varnauld** [comment]
Thank you @EVOsparkLP for your feedback - I will let @XiangyuL-Microsoft  follow up (Tracking n#40152365)

**Comment by XiangyuL-Microsoft** [comment]
@EVOsparkLP Hi, It's noticed that the ARFoundation version in this case is a bit older than the version in #284. Are you able to upgrade the ARFoundation to 4.17 or higher and retry?

**Comment by EVOsparkLP** [comment]
@XiangyuL-Microsoft I checked with 4.1.10 which is currently the "verified" version for Unity 2020.3. I also tried 4.2.3 which is the latest version available for Unity 2020. I guess the version mentioned in #284 is not 4.17 but 4.1.7 which was the latest version at the time the issue was created. So to me it seems everything is already up to date.

Maybe it's worth to mention that the anchor creation has no impact in this issue. No need to go through all the steps of the sample scene: You can directly go from StartSession to StopSession to reproduce the issue:
```
case AppState.DemoStepStartSession:
    currentAppState = AppState.DemoStepBusy;
    await CloudManager.StartSessionAsync();
    //currentAppState = AppState.DemoStepCreateLocalAnchor;
    currentAppState = AppState.DemoStepStopSession;
    break;
```

**Comment by XiangyuL-Microsoft** [comment]
@EVOsparkLP

I reviewed @msftradford Radford's workaround in #284.
It seems that you should call certain methods in such a sequence to avoid crash:

1.  ARSession.Reset();
2. await CloudManager.ResetSessionAsync();
3. CloudManager.DestroySession();

But in the top post of this thread, I see you called 3 before 1 and missing 2. Could you please try the above sequence to see if it help?

Thank you.

**Comment by XiangyuL-Microsoft** [comment]
> = FindObjectOfType<ARSession>();

Please try modifying the "new" code like this:

```
                    ARSession arSession = FindObjectOfType<ARSession>();

                    Debug.Log("Reset AR Foundation Session");
                    arSession.Reset();
                    Debug.Log("Reset Spatial Anchors Session");
                    await CloudManager.ResetSessionAsync();
                    Debug.Log("Destory Spatial Anchors Session");
                    CloudManager.DestroySession();

```

**Comment by EVOsparkLP** [comment]
@XiangyuL-Microsoft 
Sorry for the late response - during the weekend I did not have access to my Mac building machine...

I tried the code suggested by you but the app still crashes when the ARSession.Reset() is called. I tested a lot of combinations of stopping, resetting and destroying the sessions but I haven't found a solution where the app does not crash during the ARSession.Reset when a CloudManager.Session was running before.

Is there any other recommended solution to pause and reset the ARSession? Currently my workaround is to stay on version 2.10.2 which does not have this issue.

**Comment by XiangyuL-Microsoft** [comment]
@EVOsparkLP 

Hi, are you "test" the combinations on the Official Azure Spatial Anchors Demo app? Or else on your own app? If it is on your own app, is it possible for you to share the project and code with us for investigation? Thank you very much.

BTW, I can share the local modified Demo project files on my Mac with you to see if it works on your side.

**Comment by XiangyuL-Microsoft** [comment]
@EVOsparkLP

There are two GOOD news I want to bring to you:

1. By building with SDK 2.12.0, I successfully reproduced the problem consistently
2. By building with the incoming SDK 2.13.0, I successfully passed all Steps without any crash.

So, we can expect the new release can resolve this problem.

**Comment by EVOsparkLP** [comment]
@XiangyuL-Microsoft 
That sounds great! When will version 2.13.0 be released? I will have a look at it as soon as it is available. Thanks for you support.

**Comment by XiangyuL-Microsoft** [comment]
> @XiangyuL-Microsoft That sounds great! When will version 2.13.0 be released? I will have a look at it as soon as it is available. Thanks for you support.

We are working the final work before release it, I'll let you know once it was released.

**Comment by Varnauld** [comment]
We just released a new version of SDK [2.13.0](https://github.com/Azure/azure-spatial-anchors-samples/releases/tag/v2.13.0).
Could you try this new version of SDK to see if the problem still persists please?
Thank you very much.

**Comment by EVOsparkLP** [comment]
I tried version 2.13.0 and the issue does not occur anymore. Seems to be fixed, thanks.
